src/v/redpanda/application.cc

```diff
@@ -1247,6 +1257,21 @@ void application::wire_up_bootstrap_services() {
       })
       .get();
 
+    // Hook up local_monitor to update storage_resources when disk state changes
+    _storage_disk_notification
+      = storage_node.local().register_disk_notification(
```

Comment:
nit: just a thought, could we forgo exposing notifications in the storage layer and instead have registration and unregistration be a part of start and stop for node::api? Or is there an upcoming idea to grow the storage notification system to cover more than just this space reporting (e.g. handling disk failures, handling corruptions, etc)?

Reply:
I'm anticipating other subsystems (e.g. cloud_storage cache service) needing notifications too